pliers/transformers/api/base.py

```diff
@@ -2,37 +2,39 @@
 
 from pliers import config
 from pliers.transformers import Transformer
-from pliers.utils import isiterable, EnvironmentKeyMixin
+from pliers.utils import isiterable, EnvironmentKeyMixin, listify
 
 
 class APITransformer(Transformer, EnvironmentKeyMixin):
 
-    def _iterate(self, stims, validation='strict', *args, **kwargs):
-        # Check if we are trying to transform a large number of stimuli
-        if not config.get_option('allow_large_jobs'):
-            stims = list(stims)  # TODO: better way to do this?
-            if len(stims) > 100:
-                raise ValueError("Attempted to run an API transformation "
-                                 "on %d stims, aborting. To allow "
-                                 "transformation, change config option "
-                                 "'allow_large_jobs' to True." % len(stims))
-
-        return super(APITransformer, self)._iterate(stims, *args, **kwargs)
+    def __init__(self, **kwargs):
+        self.transformed_stim_count = 0
+        super(APITransformer, self).__init__(**kwargs)
 
     def _transform(self, stim, *args, **kwargs):
         # Check if we are trying to transform a large amount of data
+        self.transformed_stim_count += len(listify(stim))
```

Comment:
This will still turn any generator into a list though, which defeats the (memory-saving) point of using generators. (I also think it might consume `stim` if it's a generator, so worth checking to make sure it actually works in that case.)

On further thought, I'm not sure what the best way to handle this is. One possibility is to increment the counter only *after* each transformation is applied. The obvious downside of that is that we risk running 100 transformations and only then raising an exception, which doesn't seem wise.

Alternatively, we could check type here (I think I added an `isgenerator` utility to `utils`) and pre-emptively check for size only if `list` is *not* a generator (and if it is, fall back on applying the transformation anyway). I don't see any way around incrementing the counter in real time for generators, because the way pliers works at the moment, the graph is expanded dynamically, so I don't think there's any way to determine ahead of time how many transformations are going to happen.

I suppose it wouldn't be unreasonable to disable the `allow_large_jobs` option in the case that the user has generators enabled (with a warning or exception if both options are set to `True`). I figure if you've actively decided to use generators, you probably expect to be running large jobs, as generators only help you in the large-memory case anyway. So maybe the path of least resistance is to leave things as-is but issue a warning if needed.

Reply:
I will need to triple check this, but I think at this point (in `_transform`), there is no way `stim` is a generator (unless `listify(video)` lists out the frames) since `transform` takes care of iteration. This line of code is necessary for batch transformers, where we already made the concession to listifying prior to transforming.

Reply:
Ohhhhh, right. Makes sense. In that case, this works for me. :)